api/v1alpha1/clientconfig_types.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ClientConfigSpec defines the desired state of KubeConfig
+type ClientConfigSpec struct {
```

Comment:
To allow defining the secret**s** that we wish to use and to allow some additional parameters (I only added cluster name override, but I guess other connection parameters could be required).

Reply:
For example, I could imagine as setting parameters such as "AllocatableMemory", CPU restrictions etc which could limit what you can deploy to remote clusters. Or give annotations / labels, such as "aws-region", "oceanic-datacenter". Or why not more restrictions like affinity / tolerations, but to pods. "If dc deployed to us-east-1, don't allow deploying second copy to us-east-2, only us-west-n".

Reply:
In general I like the idea of additional parameters, but I am not sure about the examples. Wouldn't we want to specify things like affinity and tolerations in the K8ssandraCluster object?

Reply:
There could be multiple K8ssandraCluster objects targetting the same clusters. Lets say there's an allocation for 8 nodes in the target cluster. You could have 2 different K8ssandraClusters which could build a DC there, both taking 3 nodes for example. But a third K8ssandraCluster with 3 nodes could no longer use that datacenter since it's full.

The other defaults could be something like cluster specific storageClass settings. For example, cluster-1 would use storageClass: premium-ssd ; cluster-2 would use storageClass: ultra. When you create the K8ssandraCluster, you could only define "400GB" and let the storageClass be modified by the target cluster's definitions. Obviously could be defined in the K8ssandraCluster each time, but we could simplify things for user (or if there's RBAC issues to make sure that some people are not allowed to define what disks they can use..)

.. or, Ingress/Egress settings if we're not using a single network in the future. Additional labels / annotations per cluster?

Reply:
But also, do we potentially want to keep status of the target cluster? For example to indicate if the connection is down.

Reply:
>do we potentially want to keep status of the target cluster?

I don't know. If we do at any point, we probably ought look into https://github.com/kubernetes-sigs/cluster-api.

To close the loop on the discussion around the `ClientConfig` type, you addressed my immediate questions. The rest of the discussion probably should live outside of the PR, maybe the forum ;-)